chain/ethereum/src/block_stream.rs

```diff
@@ -629,7 +629,7 @@ where
         let head_ptr_opt = self.chain_store.chain_head_ptr()?;
         let subgraph_ptr = self.subgraph_store.block_ptr(self.subgraph_id.clone())?;
 
-        if head_ptr_opt != subgraph_ptr {
+        if head_ptr_opt.is_none() || subgraph_ptr.is_none() || head_ptr_opt != subgraph_ptr {
```

Comment:
Maybe you found it more readable this way, but checking if `subgraph_ptr` is `None` is redundant with the other two checks. If `head_ptr_opt` is `None` then it's redundant with `head_ptr_opt.is_none()`. If `head_ptr_opt` is `Some` it is redundant with `head_ptr_opt != subgraph_ptr`.

Reply:
Yes, I just liked the symmetry better since the redundant check gives you a little less pause when reading that. I think the condition should be changed to have the common case `head_ptr_opt != subgraph_ptr` first though